src/parser/jobs/ast/modules/EarthlyStar.tsx

```diff
@@ -96,6 +112,11 @@ export default class EarthlyStar extends Module {
 			}))
 		}
 
+		// If they stopped using Star at any point in the fight, this'll calculate the drift "accurately"
+		if (this.parser.fight.end_time - this.lastUse > (this.data.actions.EARTHLY_STAR.cooldown * 1000)) {
+			this.totalHeld += (this.parser.fight.end_time - (this.lastUse + (this.data.actions.EARTHLY_STAR.cooldown * 1000)))
+		}
```

Comment:
Is it not accurate already? The held bump could prob be a method of it's own so you end up with less funky math like this. Again, won't block on this, but the whole module could use a cleanup in this regard.

Reply:
I think some form of logic for this is required.
Say they use Earthly Star once at the beginning of the fight, and never again. `onPlace` won't be called again, so nothing will increase the `totalHeld` variable, so at the end of the fight we'll need to see how many more they could have cast.

Reply:
So?

Reply:
Maybe I'm missing your point. It sounded like you thought this code wasn't required based on "Is it not accurate already?"

Reply:
I get what it does, my point was about fixing the initial drift conditional and collapsing the 3 diff calculations into a single method that can be called in onPlace and onComplete to avoid duplicating complex-looking math and making it easier to maintain.